go/vt/mysqlctl/schema.go

```diff
@@ -579,13 +579,7 @@ func (mysqld *Mysqld) ApplySchemaChange(ctx context.Context, dbName string, chan
 // defined PRIMARY KEY then it may return the columns for
 // that index if it is likely the most efficient one amongst
 // the available PKE indexes on the table.
-func (mysqld *Mysqld) GetPrimaryKeyEquivalentColumns(ctx context.Context, dbName, table string) ([]string, string, error) {
-	conn, err := getPoolReconnect(ctx, mysqld.dbaPool)
```

Comment:
Did the functionality change for this PR require a change in how we acquire a connection here?

Reply:
When I first created this I had a `mysqlctl.Mysqld` instance readily available in the callsite. In working on this, I realized that I should have changed it when using it in vstreamer as I created a new instance there, which is heavy as it creates conn pools etc (and I wasn't properly calling close in a defer 🤦‍♂️), so I changed it here to use a callback to talk to the DB instead. It's much lighter and is far better (and is an already established pattern in the mysqlctl package).

Reply:
If we adopt this pattern, then we will find ourselves passing `exec` functions in a large amount of functions - not saying it's wrong - just thinking of the code/signature impact.

Reply:
It's already a pattern (in that file/package and elsewhere). I agree that it's not one which should be used w/o good reason.